### PR TITLE
Fields type and the general query

### PR DESCRIPTION
**Pull request by MohammedYehia**
- Add a new type to DB schema. 
- Add a new column that checks if the profile is complete.
- Change contract location to contract region with the type region and add the restrict NOT NULL. 
- Add the basic and general query executor function.

### CODE REVIEW

File: server/database/db.sql [hunk 1]
```diff
@@ -1,7 +1,7 @@
 BEGIN;
 
 DROP TABLE IF EXISTS social_enterprise_basic, social_enterprise_details, contract, policy, social_impact_list; 
-DROP TYPE IF EXISTS regions, trade_type, turnover_size, social_mission, uk_cities;
+DROP TYPE IF EXISTS regions, trade_type, turnover_size, contract_size, social_mission, uk_cities;
 
 CREATE TYPE regions AS ENUM(
```
Comment (sultanassi95) [inline server/database/db.sql]: use uppercase letters for Database types ==> `REGIONES` .. do that for the rest of the types you have

Comment (MohammedYehia) [reply server/database/db.sql]: I would be very grateful if you could reference me to a link for the conventions about this.
As the official Postgres [documentation](https://www.postgresql.org/docs/9.1/static/datatype-enum.html) use lowercase with the definition for ENUM type.

File: server/database/db.sql [hunk 2]
```diff
@@ -104,8 +122,9 @@ CREATE TABLE contract(
     open_for_splitting BOOLEAN DEFAULT FALSE,
     active_state BOOLEAN DEFAULT TRUE, 
     company_name VARCHAR UNIQUE NOT NULL,
-    location uk_cities,
+    contract_region regions NOT NULL,
     post_code VARCHAR NOT NULL,
     created_at TIMESTAMP NOT NULL DEFAULT CURRENT_TIMESTAMP,
```
Comment (RamyAlshurafa) [inline server/database/db.sql]: remove last comma

File: server/database/db.sql (hunk 1 above)
Comment (sultanassi95) [reply server/database/db.sql]: you absolutely nailed this one :clap:

**Pull request merged**